src/devices/Common/System/Device/DelayHelper.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Device
+{
+    /// <summary>
+    /// Helpers for short waits.
+    /// </summary>
+    internal static class DelayHelper
+    {
+        // GetTimestamp() currently can take ~300ns. We hope to improve this to get better
+        // fidelity for very tight spins.
+        //
+        // SpinWait currently spins to approximately 1μs before it will yield the thread.
+
+        /// <summary>
+        /// Delay for at least the specified <paramref name="microseconds"/>.
+        /// </summary>
+        /// <param name="microseconds">The number of microseconds to delay.</param>
+        /// <param name="allowThreadYield">True to allow yielding the thread.</param>
+        public static void DelayMicroseconds(int microseconds, bool allowThreadYield = true)
+        {
+            long start = Stopwatch.GetTimestamp();
+            long end = start + (microseconds * Stopwatch.Frequency / 1_000_000);
```

Comment:
The add may overflow. It is probably safer to compute the duration in ticks and in the loop check the elapsed ticks from the start against the desired duration. Same for `DelayMilliseconds()`.

Reply:
I was trying to minimize the amount of work done in the loop. I'll redo it.

Reply:
I think if you just keep on subtracting current time and start time (or current and end time) you can safely overflow since it will cancel each other

Reply:
> I think if you just keep on subtracting current time and start time

I'm not following

Reply:
Ok, updated. Please check my brain. XD